internal/webhook/template_webhook.go

```diff
@@ -187,7 +187,9 @@ func (*ServiceTemplateValidator) Default(_ context.Context, obj runtime.Object)
 	if !ok {
 		return apierrors.NewBadRequest(fmt.Sprintf("expected ServiceTemplate but got a %T", obj))
 	}
-	setHelmChartDefaults(template.GetHelmSpec())
+	if template.Spec.Helm != nil {
+		setHelmChartDefaults(template.GetHelmSpec())
+	}
```

Comment:
The `setHelmChartDefaults` function already [returns if helmSpec == nil](https://github.com/k0rdent/kcm/blob/ae7a42ff853e3b12e67ca8d40b4ebda03a6442b9/internal/webhook/template_webhook.go#L322-L324).